src/index.js

```diff
@@ -1,374 +1,555 @@
-        throw new Error(errors.VENDOR_DOWN)
+  async autoLoginCheckAuthenticated() {
+    this.log('info', 'autoLoginCheckauthenticated starts')
+    await waitFor(
+      () => {
+        return Boolean(
+          document.querySelector(
+            'button[data-test="account-disconnect-button"]'
+          )
+        )
+      },
+      {
+        interval: 1000,
+        timeout: {
+          milliseconds: 30000,
+          message: new TimeoutError(
+            'autoLoginCheckAuthenticated timed out after 30000ms'
+          )
+        }
       }
-    }
+    )
+    return true
   }
 
-  async testSession() {
-    if (!this._jar._jar.toJSON().cookies.length) {
-      return false
+  async findAndSendCredentials(loginField, passwordField) {
+    this.log('info', 'findAndSendCredentials starts')
+    let userLogin = loginField.value
+    let userPassword = passwordField.value
+    const userCredentials = {
+      email: userLogin,
+      password: userPassword
     }
-    const request = this.requestFactory({
-      cheerio: false,
-      json: false
+    return userCredentials
+  }
+
+  async showLoginFormAndWaitForAuthentication() {
+    log.debug('showLoginFormAndWaitForAuthentication start')
+    await this.setWorkerState({ visible: true })
+    await this.runInWorkerUntilTrue({
+      method: 'waitForAuthenticated'
     })
-    log('info', 'Test the validity of old session')
-    try {
-      const resp = await request({
-        uri: 'https://www.oui.sncf/espaceclient/commandes-en-cours',
-        resolveWithFullResponse: true,
-        followRedirect: false,
-        followAllRedirects: false
-      })
-      if (
-        resp.statusCode === 200 &&
-        resp.request.uri.href ===
-          'https://www.oui.sncf/espaceclient/commandes-en-cours'
-      ) {
-        log('info', 'Session valid')
-        return true
-      } else {
-        log('info', 'Session invalid')
-        return false
+    await this.setWorkerState({ visible: false })
+  }
+
+  async getUserDataFromWebsite() {
+    this.log('info', '🤖 getUserDataFromWebsite')
+    await this.navigateToPersonnalInfos()
+    await this.waitForElementInWorker(
+      'a[href="https://tgvinoui.sncf/compte/informations/modification-identifiant"]'
+    )
+    await this.runInWorker('getIdentity')
+    if (this.store.userIdentity.email) {
+      return {
+        sourceAccountIdentifier: this.store.userIdentity.email
       }
-    } catch (err) {
-      log('info', 'Getting error during testing, Session invalid')
-      return false
+    } else {
+      throw new Error('No user data identifier, the konnector should be fixed')
     }
   }
 
-  async getPastOrders() {
-    const $ = await this.getPastOrderPage()
-    return parseOrderPage($)
+  async fetch(context) {
+    this.log('info', '🤖 fetch')
+    if (this.store.userCredentials) {
+      await this.saveCredentials(this.store.userCredentials)
+    }
+    await this.saveIdentity(this.store.userIdentity)
```

Comment:
nit: The identity given to saveIdentity should be in the form : {contact: {...}} or else you get a warning : https://github.com/konnectors/libs/blob/5c57c1658e841d12120edbb8da198d3f294bb9d4/packages/cozy-clisk/src/launcher/saveIdentity.js#L40-L45